### ISSUE TITLE
Builder dependencies

### ISSUE THREAD

**Issue by adam-mcdaniel** [adam-mcdaniel/rusty-ci#58]
We should be able to have builders depend on one another, meaning only having some builders run after the depending builder has succeeded.

This must be done in the scheduler, because the builders are only meant to be atomic tasks that can be put together like legos in schedulers.

Maybe we have the scheduler `builders` section be a list of lists?

It could work as follows

```yaml
builders:
  # This is the first list of steps to run, it only has clang-format though
  - - clang-format
  # This next list of steps will only happen after the first list has succeeded
  - - build-test
    - ctest
```

However, this is kind of ugly and I can see users quickly typing out a malformed section and getting very confused as to why it's not working properly.

We might not care though.

I'll keep this in the back of my mind.

**Comment by adam-mcdaniel** [comment]
You can implement something like this using buildbot's Dependent scheduler.

Example:
```python
from buildbot.plugins import schedulers
tests = schedulers.SingleBranchScheduler(name="just-tests",
                                         treeStableTimer=5*60,
                                         builderNames=["full-linux",
                                                       "full-netbsd",
                                                       "full-OSX"])
package = schedulers.Dependent(name="build-package",
                               upstream=tests, # <- no quotes!
                               builderNames=["make-tarball", "make-deb",
                                             "make-rpm"])
c['schedulers'] = [tests, package]
```


With this, you could have a `depends` section in schedulers like this:

```yaml
scheduler1:
  # scheduler1 depends on scheduler2
  depends:
    - scheduler2
  builders:
    - builder1
```

The depends section would replace the `triggers` and `branch` sections in functionality

**Comment by adam-mcdaniel** [comment]
Fixed with #64
